### PR TITLE
cruft: init at 2.15.0

### PR DESCRIPTION
**Pull request by avh4**
## Description of changes

https://cruft.github.io/cruft/

Cruft is a command-line tool that builds on top of cookiecutter (already packaged in nixpkgs) to allow projects created with cookiecutter to merge in changes from the base template.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/cruft/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ lib, buildPythonPackage, fetchPypi, poetry-core
+, click, cookiecutter, gitpython, importlib-metadata, toml, typer
+}:
+
+buildPythonPackage rec {
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/cruft/default.nix:5]: Why not `buildPythonApplication`?

**Comment by avh4** [comment]
Ah, I didn't know about that at the time. I will fix it.

**Comment by aacebedo** [comment]
@avh4 can you fix the comment ? I would like to use this tool.

**Comment by aacebedo** [comment]
Up.

I can complete the PR if you don't have time to do it

**Pull request closed without merging**